askama/src/lib.rs

```diff
@@ -502,6 +502,10 @@ pub trait Template {
     fn extension() -> Option<&'static str>
     where
         Self: Sized;
+    fn dyn_size_hint(&self) -> usize;
```

Comment:
I would say maybe we can just name this `size_hint()` for now? Not sure if the conflict causes issues in the common case.

Reply:
If they have the same name then importing `SizedTemplate` and trying to use the receiverless `size_hint()` will give a "multiple applicable items in scope" compile error if `Template` is also imported.

Reply:
Yeah, I think I'm okay with that.

Reply:
Doesn't it basically make the receiverless `size_hint()` unusable? Or is there a way to disambiguate?

(Giving `extension()` the same treatment causes the same compile error in `testing/tests/simple.rs`.)

Reply:
I think you can do something like `SizedTemplate::extension(&t)` or `<t as SizedTemplate>::extension()`.

Reply:
ah yes, it works. I've updated the PR.